crates/goose/src/agents/truncate.rs

```diff
@@ -144,20 +136,20 @@ impl Agent for TruncateAgent {
             debug!("user_message" = &content);
         }
 
-        // Update conversation history for the start of the reply
         let mut messages = self
-            .prepare_inference(
+            .enforce_ctx_limit(
```

Comment:
i still think we want this to happen not just before the loop but during each iteration of the loop as well? 

a few ways to do this but i'd imagine putting it just before 
```
let (response, usage) = capabilities.provider().complete(
```
is the simplest.

Without that change, if subsequent tool calls push the model past the limit it won't have a chance to truncate to continue the loop?